plugins/inputs/logfile/logfile_test.go

```diff
@@ -438,6 +438,10 @@ func TestLogsFileAutoRemoval(t *testing.T) {
 		t.Errorf("Wrong log found from 2nd file: \n% x\nExpecting:\n% x\n", e.Message(), logEntryString)
 	}
 
+	//Sleep for avoiding race condition between opening tmpfile2 to delete tmpfile1 with auto_removal and opening tmpfile1
+	//to check it exist
+	time.Sleep(2 * time.Second)
```

Comment:
I would rather not add an additional sleep here. How flaky is this test? Is it worth addressing directly in this PR? 

Rather than adding a 2 second wait here, would it work to create a WaitGroup, call `Add()` at the top, then inside the goroutine that creates tmpfile2, call `Done()`, and then replace this sleep with a `Wait()` call?

Reply:
It is a little bit overkill for adding WaitGroup here. However, it's fine for me to add it since it's better to use it than sleep. However, if adding a WaitGroup is a better choice, then addressing this in another PR is fine for me though. For your information, 1 out of 3 `make test` would fail it.